deployment/app/src/main/java/com/rk/amii/database/DBHandler.java

```diff
@@ -457,7 +483,92 @@ public ArrayList<PhotoModel> getPhotoInfo(Integer assessmentId) {
     @Override
     public void onUpgrade(SQLiteDatabase db, int oldVersion, int newVersion) {
         db.execSQL("DROP TABLE IF EXISTS " + SITES_TABLE_NAME);
+        db.execSQL("DROP TABLE IF EXISTS " + USER_TABLE_NAME);
         onCreate(db);
     }
 
+    /**
+     * Add a new user to the database
+     * @param username The username
+     * @param email The email address
+     * @param name The user's first name
+     * @param surname The user's surname
+     * @param organisationType The type of organisation (can be null)
+     * @param organisationName The name of the organisation
+     * @param country The user's country
+     * @param uploadPreference The user's upload preference (e.g., wifi)
+     * @return The user id
+     */
+    public long addOrUpdateUserProfile(String username, String email, String name, String surname,

```

Comment:
Please create a separate PR for this.

Reply:
@dimasciput It is related since the upload preference is located User Profile. So they use same endpoint, which is endpoint to update/get user profile.